hummingbot/connector/exchange/xrpl/xrpl_exchange.py

```diff
@@ -0,0 +1,1430 @@
+    def trading_pairs_request_path(self):
+        return ""
+
+    @property
+    def check_network_request_path(self):
+        return ""
+
+    @property
+    def trading_pairs(self):
+        return self._trading_pairs
+
+    @property
+    def is_cancel_request_in_exchange_synchronous(self) -> bool:
+        return False
+
+    @property
+    def is_trading_required(self) -> bool:
+        return self._trading_required
+
+    @property
+    def node_url(self) -> str:
+        return self._wss_node_url
+
+    @property
+    def second_node_url(self) -> str:
+        return self._wss_second_node_url
+
+    @property
+    def auth(self) -> XRPLAuth:
+        return self._auth
+
+    def supported_order_types(self):
+        return [OrderType.LIMIT, OrderType.LIMIT_MAKER, OrderType.MARKET]
+
+    def _is_request_exception_related_to_time_synchronizer(self, request_exception: Exception):
+        # API documentation does not clarify the error message for timestamp related problems
+        return False
+
+    def _is_order_not_found_during_status_update_error(self, status_update_exception: Exception) -> bool:
+        # TODO: implement this method correctly for the connector
+        # The default implementation was added when the functionality to detect not found orders was introduced in the
+        # ExchangePyBase class. Also fix the unit test test_lost_order_removed_if_not_found_during_order_status_update
+        # when replacing the dummy implementation
+        return False
+
+    def _is_order_not_found_during_cancelation_error(self, cancelation_exception: Exception) -> bool:
+        # TODO: implement this method correctly for the connector
+        # The default implementation was added when the functionality to detect not found orders was introduced in the
+        # ExchangePyBase class. Also fix the unit test test_lost_order_removed_if_not_found_during_order_status_update
+        # when replacing the dummy implementation
+        return False
+
+    def _create_web_assistants_factory(self) -> WebAssistantsFactory:
+        pass
+
+    def _create_order_book_data_source(self) -> OrderBookTrackerDataSource:
+        return XRPLAPIOrderBookDataSource(
+            trading_pairs=self._trading_pairs, connector=self, api_factory=self._web_assistants_factory
+        )
+
+    def _create_user_stream_data_source(self) -> UserStreamTrackerDataSource:
+        return XRPLAPIUserStreamDataSource(auth=self._auth, connector=self)
+
+    def _get_fee(
+        self,
+        base_currency: str,
+        quote_currency: str,
+        order_type: OrderType,
+        order_side: TradeType,
+        amount: Decimal,
+        price: Decimal = s_decimal_NaN,
+        is_maker: Optional[bool] = None,
+    ) -> TradeFeeBase:
+        # TODO: Implement get fee, use the below implementation
+        # is_maker = is_maker or (order_type is OrderType.LIMIT_MAKER)
+        # trading_pair = combine_to_hb_trading_pair(base=base_currency, quote=quote_currency)
```

Comment:
will be good to have this implementation

Reply:
Thanks, I will work on this later.